### ISSUE TITLE
ResNet architectures on Cifar10 do not follow Kaiming He

### ISSUE THREAD

**Issue by kudkudak** [facebookresearch/mixup-cifar10#3]
I think Resnets models (Resnet-18, Resnet-110, etc.) do not match https://arxiv.org/abs/1603.05027. The reason is that Kaiming He used a different architecture (under the same name) for CIFAR-10 compared to ImageNet. In particular Resnet-18 should have ~0.3M params, while the one used in the repository seems to have ~10M. See Tab. 6 and bottom of page 7 from https://arxiv.org/abs/1512.03385:

![image](https://user-images.githubusercontent.com/1571945/44103474-3da2657e-9fe4-11e8-8828-9073ce697734.png)

This is not a bug of course, but it might be a bit misleading.

**Comment by kudkudak** [comment]
I am not sure if the Resnet even follows the ImageNet version. Before softmax there should be a final ReLU -> BN: https://github.com/KaimingHe/resnet-1k-layers/blob/master/resnet-pre-act.lua#L118, which if I am looking correctly is absent here https://github.com/facebookresearch/mixup-cifar10/blob/master/models/resnet.py#L166.

**Comment by PabloRR100** [comment]
Actually, the authors only use 3 layers for ResNets on CIFAR-10. 
I am trying to work with ensembles of ResNets on CIFAR-10 so this is my implementation on PyTorch for ResNets on CIFAR-10:

https://github.com/PabloRR100/ResNets_Ensembles/blob/master/resnetsCIFAR10.py

**Comment by ynd** [comment]
Hi @kudkudak,

Thanks for your careful read of the paper and code. Indeed, the Resnet paper (https://arxiv.org/abs/1603.05027) uses two slightly different architectures with the same name. The name refers to the number of layers in the network - a name that captures every implementation detail would be pretty complicated (especially when cross-validation can change these specifics, such as the width of the layer)

We use the same convention in our work, and we have made sure to run all our baselines. We appreciate that leaves some ambiguity and so we have provided code for reference and reproducibility of the paper. That's the main purpose of this repo, and changing the code in such a way now would be detrimental to this purpose.
